File: app/providers/parsers/markdown.py
from pathlib import Path

from app.providers.parsers.base import DocumentParser, ParsedDocument


class MarkdownParser(DocumentParser):
    @property
    def name(self) -> str:
        return "markdown"

    def supports(self, filename: str, mime_type: str | None = None) -> bool:
        suffix = Path(filename).suffix.lower()
        return suffix in {".md", ".txt", ".markdown"}

    async def parse(self, file_bytes: bytes, *, filename: str) -> ParsedDocument:
        text = file_bytes.decode("utf-8", errors="replace").replace("\r\n", "\n").strip()
        suffix = Path(filename).suffix.lower()
        source_type = "markdown" if suffix in {".md", ".markdown"} else "text"
        return ParsedDocument(
            content=text,
            source_type=source_type,
            metadata={"parser": self.name},
        )

File: frontend/src/utils/markdown.ts
/** 清理 chunk 中残留的 docx 书签/HTML 标签，避免 Markdown 渲染成可见文本。 */
export function sanitizeChunkMarkdown(content: string): string {
  return content
    .replace(/<a\s+id="[^"]*"\s*><\/a>\s*/gi, "")
    .replace(/<br\s*\/?>/gi, "\n");
}
